src/RL/representation/default/cp_layer/accessors.jl

```diff
@@ -137,14 +137,37 @@ Base.zero(::Type{CPLayerGraph}) = CPLayerGraph()
 Base.reverse(g::CPLayerGraph) = g
 
 function LightGraphs.SimpleGraph(cplayergraph::CPLayerGraph)
-    graph = Graph(edges(cplayergraph))
-    n = nv(cplayergraph)
-    if nv(graph) < n
-        add_vertices!(graph, n - nv(graph))
+    graph = LightGraphs.Graph(LightGraphs.edges(cplayergraph))
+    n = LightGraphs.nv(cplayergraph)
+    if LightGraphs.nv(graph) < n
+        LightGraphs.add_vertices!(graph, n - LightGraphs.nv(graph))
     end
     return graph
 end
 
 function LightGraphs.adjacency_matrix(cplayergraph::CPLayerGraph)
-    return adjacency_matrix(Graph(cplayergraph))
+    return LightGraphs.adjacency_matrix(LightGraphs.Graph(cplayergraph))
 end
+
+function adjacency_matrices(cplayergraph::CPLayerGraph)
+    g = LightGraphs.Graph(cplayergraph) # Update the graph with the new information
+    nvar = cplayergraph.numberOfVariables
+    ncon = cplayergraph.numberOfConstraints
+    nval = cplayergraph.numberOfValues
+    contovar = zeros(ncon, nvar)
+    valtovar = zeros(nval, nvar)
+    for (i, node) in enumerate(cplayergraph.idToNode)
+        if isa(node, ConstraintVertex)
+            neighbors = LightGraphs.outneighbors(g, i)
+            for neighbor in neighbors
+                contovar[i, neighbor - ncon] = 1
+            end
+        elseif isa(node, ValueVertex)
+            neighbors = LightGraphs.outneighbors(g, i)
+            for neighbor in neighbors
+                valtovar[i - ncon - nvar, neighbor - ncon] = 1
+            end
+        end
+    end
+    return contovar, valtovar

```

Comment:
Did we tested this ?

Reply:
Yes, we have unit tests for this in test/heterogeneousstaterepresentation.jl.